Fix Parametric Map compliance issues

hackermd opened the pull request:
I realized that the Parametric Map IOD requires the Modality LUT Transformation step in the grayscale rendering pipeline to be an identity transformation (i.e., it requires Identity Pixel Value Transformation rather than Pixel Value Transformation): https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_A.75.5.html

That also means only unsigned integer values can be stored. They can still be mapped to a different range of "real world" values though using Real World Value Mapping.

Also fixed several other compliance issues.

cgorman commented on src/highdicom/pm/enum.py:
```diff
@@ -0,0 +1,45 @@
+from enum import Enum
+
+
+class ImageFlavorValues(Enum):
```
Should we consider including the [CT- and MR-specific defined terms from C.8.15.2.1.1.3 and C.8.13.1.1.1.3](https://dicom.innolitics.com/ciods/parametric-map/parametric-map-image/00080008)? (also for the derived pixel contrast values)

hackermd replied:
Addressed via 7216af5c10b05131f42f9ec1a5462f0ede2ebe1f

cgorman reviewed (commented):
Looks good, I just had a comment on the image value enums but that's it.

hackermd commented:
@cgorman I added a unit test for Parametric Map that will test the dimension index lookup in case spatial coordinates are not preserved. That part of the code was previously not covered.

I also went ahead and fixed #135.

hackermd commented:
@cgorman also realized that #122 was actually not yet implemented. Fixed that and added a unit test.
